docs/cache-server.md

```diff
@@ -0,0 +1,125 @@
+### Purpose
+
+The primary purpose of the cache server is to support cross-shard communication.
+Shards need to communicate with each other in order to enable, among other things, migration and replication features.
+Direct shard-to-shard communication is not feasible in a larger setup.
```

Comment:
```suggestion
Direct shard-to-shard communication is not feasible in a larger setup as it scales with n^2.
```

Reply:
(or some other way to indicate *why* it's not feasible)

Reply:
updated.